client/cli/src/commands/run_cmd.rs

```diff
@@ -404,7 +404,7 @@ pub fn is_node_name_valid(_name: &str) -> std::result::Result<(), &str> {
 		return Err("Node name should not contain invalid chars such as '.' and '@'")
 	}
 
-	let invalid_patterns = r"(https?:\\/+)?(www)+";
+	let invalid_patterns = r"^https?:\/\/";
```

Comment:
`let invalid_patterns = r"^https?:\\/\\/";`

Instead?

Reply:
I think you don't even need to escape the `/` when using raw string, so just `r"^https?://"` should work. Also none of the tests would catch this since the URLs also include a `.` so this condition is never tested.

Reply:
https://github.com/paritytech/substrate/pull/14593